### PR TITLE
Add PNG encoder

### PR DESCRIPTION
**Pull request by benmcmorran**
Because why not...

I wrote [this encoder](https://gist.github.com/benmcmorran/815fd6505ed2b234c3c27b9a0d67226b) a while ago on a whim, and this seems like a great project to add it to. To use it, just specify the `use_png` variable as part of the CMake invocation. Unfortunately, there's currently [no way to write binary files from CMake](https://gitlab.kitware.com/cmake/cmake/-/issues/21878), so at the very end of the process it currently falls back to PowerShell. The generated PNGs have no compression but are valid. I checked in a new version of the `render.png` that was generated with this encoder. It should have exactly the same pixel content but is significantly larger due to the lack of compression.

I also changed the ray tracer to run in [script mode](https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-a-script) instead of as an actual build. This lets CMake skip all the complier and build system checks that it normally does.

### CODE REVIEW

**Comment by 64** [comment]
lmao

File: README.md [hunk 1]
```diff
@@ -9,11 +9,17 @@ A simple ray tracer written in pure CMake. Inspired by [raytracer.hpp](https://g
 The ray tracer writes its output to `stderr`, so you can use it with:
 
 ```shell
-cmake . -Dimage_width=64 -Dimage_height=64 -Dnum_procs=4 2> image.ppm
+cmake -Dimage_width=64 -Dimage_height=64 -Dnum_procs=4 -P CMakeLists.txt 2> image.ppm
 ```
 
 Which writes the output to `image.ppm`. Then use an image viewer capable of opening PPM files (or [this](http://www.cs.rhodes.edu/welshc/COMP141_F16/ppmReader.html)) to view.
 
+Alternatively, the ray tracer can directly write PNG files (PowerShell required for now). Use the `use_png` variable to set the name of the PNG file without the extension. The example below will write output to `render.png`.
```
Comment (64) [inline README.md]: Feel free to give yourself some credit here (write something like 'courtesy of @benmcmorran' or 'this feature was implemented by @benmcmorran')

**Comment by 64** [comment]
lgtm otherwise, thanks

File: README.md (hunk 1 above)
Comment (benmcmorran) [reply README.md]: Added, thanks!

**Comment by 64** [comment]
excellent stuff, thanks again

**Pull request merged**